pyupgrade/_plugins/typing_pep563.py

```diff
@@ -0,0 +1,90 @@
+import ast
+import functools
+from typing import Iterable
+from typing import MutableMapping
+from typing import Optional
+from typing import Tuple
+from typing import Union
+
+from tokenize_rt import Offset
+
+from pyupgrade._ast_helpers import ast_to_offset
+from pyupgrade._data import register
+from pyupgrade._data import State
+from pyupgrade._data import TokenFunc
+from pyupgrade._token_helpers import dequote
+
+
+def _supported_version(state: State) -> bool:
+    return (
+        state.settings.min_version >= (3, 10) or (
+            'annotations' in state.from_imports['__future__']
+        )
+    )
+
+
+def replace_string_literal(
+    annotation: Optional[Union[ast.expr, ast.slice]],
+    to_replace: MutableMapping[ast.expr, str],
+) -> None:
+    if isinstance(annotation, ast.Constant):
+        if isinstance(annotation.value, str):  # pragma: no cover
+            # Not for Python 3.6, Python 3.7
```

Comment:
for version-specific no-covers, place them inside the pragma itself: `# pragma: no cover (py38+)`